iocore/eventsystem/Thread.cc

```diff
@@ -55,9 +55,8 @@ Thread::~Thread()
 }
 
 static void
-key_destructor(void *value)
+key_destructor(void *)
```

Comment:
The destructor for `pthread_key_create` is optional, so we can nuke this altogether.